### ISSUE TITLE
Autoplay counts objects twice when rewinding

### ISSUE THREAD

**Issue by Ferdi265** [ppy/osu#4597]
**Describe the bug:** 

The hitcounts (great, good, meh, miss) in an autoplay are incorrect (they are a lot too high). I have noticed the same impossibly high hitcounts with non-autoplays, but it is much easier to replicate with autoplay.

Also, when rewinding during an Autoplay, the hitcount (great, meh, miss) at the end is even higher! It seems the hitcount is not reset when rewinding, leading the already wrong hitcount to be counted multiple times.

**Screenshots or videos showing encountered issue:** 

beatmap used for demonstration: [Skuggin' Around - Kitsune^2, mapped by vipto](https://osu.ppy.sh/beatmapsets/111631#osu/290310), Easy Difficulty

## Regular Autoplay
![regular](https://i.imgur.com/mtLgHnA.jpg)

## Autoplay with Rewind
![with-rewind](https://i.imgur.com/qKvKRDU.jpg)

## Best play in the 'Global' tab
![best-play-global](https://i.imgur.com/Cdy4GTp.jpg)

**osu!lazer version:** 
Latest master at the time of writing, commit 1b7d8b7dcf8d78130c56846a0815430a5d194bf4, running on ArchLinux.

**Comment by Ferdi265** [comment]
I tested this also on the beatmap [haL1 - Hatsune Miku, mapped by AngelHoney](https://osu.ppy.sh/beatmapsets/12046#osu/45576), No Slider?! Difficulty

I chose that specific map because I suspected the bug had to do with how sliders were counted and wanted to try on a map that has no sliders.

On this map Autoplay gets the hitcount correct when not rewinding. It still counts everything twice / multiple times when rewinding though, so that seems to be a seperate issue.
